src/python/pants/backend/python/rules/repl.py

```diff
@@ -87,7 +86,10 @@ async def create_ipython_repl_request(repl: IPythonRepl, ipython: IPython) -> Re
     return ReplRequest(
         digest=merged_digest,
         args=args,
-        env={"PEX_EXTRA_SYS_PATH": ":".join(chrooted_source_roots)},
+        env={
+            "PEX_PATH": repl.in_chroot(requirements_pex_request.output_filename),
```

Comment:
Is this a refactor, or it results in behavior change?

Reply:
No behavior change. 

In `run` we don't have access to the chroot at pex build time, so we can't use the `--pex-path` cmd-line arg to embed the pex path in the PEX_INFO. Instead we do it at pex runtime via the `PEX_PATH` env var, which has the same effect. I made the same change here just to be uniform.